code-challenges.test.js

```diff
@@ -15,35 +15,115 @@
+//splice single random index values and push them into new array
+// use Math.random
+// return new array
+
+
+// const removeFirstItem = (array) => {
+//   let n = array.length
+//   let newArr =[]
+//   array.shift()
+//   while (array.length >0) {
+//     let randomnum = Math.floor(Math.random() * n--)
+//     let spliceRandomly= array.splice(randomNum, 1)
+//     newArr.push(spliceRandomly)
+// }
+// return newArr.flat()
+//
+// }
+
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
-
+// const nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// const nums2 = [109, 5, 9, 67, 8, 24]
+// // Expected output: [5, 109]
+// //
+// describe("minAndMax", () => {
+//   it ("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+//   expect(minAndMax(nums1)).toEqual(["-8","90"])
+//   expect(minAndMax(nums2)).toEqual(["5","109"])
+//   })
+// })
 
 // b) Create the function that makes the test pass.
 
+//create a function called min and max that returns the min and max values
+//minAndMax takes an array as a parameter
+// declare two variables
+// min variable takes array with spread operator and uses Math.min
+// max variable takes in array with spread operator and uses Math.max
+// return array
+
+// const minAndMax = (array) => {
+//   let nums1= [3, 56, 90, -8, 0, 23, 6]
+//   let nums2= [109, 5, 9, 67, 8, 24]
+//
+//   return Math.min([...array])
+//   return Math.max([...array])
+// }
+//   console.log(minAndMax(nums1))
+//   console.log(minAndMax(nums2))
 
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
-// a) Create a test with an expect statement using the variables provided.
 
-const testArray1 = [3, 7, 10, 5, 4, 3, 3]
-const testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
 
+// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+
+// a) Create a test with an expect statement using the variables provided.
+//
+// const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+// const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+// // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+//
+// describe("noDuplicates", () => {
+//   it ("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+//   expect(noDuplicates(testArray1,testArray2)).toEqual(["3","7","10","5","4","8","2","1"])
+//
+//   })
+// })
```

Comment:
Looks like your test is asking for numbers as strings instead of numbers. Keep an eye on the data types you are working with both input and output.